File: ruby/hamming/hamming.rb
class Hamming
  VERSION = 1

  def self.compute(s1, s2)
    raise ArgumentError, "Strands must be of equal length." if s1.length != s2.length
    empty_strands?(s1, s2) ? 0 : tally_differences(s1, s2)
  end

  def self.empty_strands?(s1, s2)
    s1.empty? && s2.empty?
  end

  def self.tally_differences(s1, s2)
    (0...s1.length).collect { |i| s1[i] != s2[i] }.count(true)
  end

end

File: javascript/hamming/hamming.js
var Hamming = function() {
	
	this.compute = function(s1,s2) {
		if (s1.length != s2.length) {
			throw "DNA strands must be of equal length.";
		}
		else if (s1 === s2) {
			return 0;
		}
		else {
			var count = 0
			for (i = 0; i < s1.length; i++) {
				if (s1[i] != s2[i]) {
				count += 1
			}
			}
			return count;
		}

	}

}

module.exports = Hamming;